tests/common/include/tests/common/path_correction.inl

```diff
@@ -7,463 +7,171 @@
 
 // Project include(s).
 #include "detray/definitions/units.hpp"
-#include "detray/detectors/create_telescope_detector.hpp"
-#include "detray/detectors/detector_metadata.hpp"
 #include "detray/masks/masks.hpp"
-#include "detray/materials/predefined_materials.hpp"
-#include "detray/propagator/actor_chain.hpp"
-#include "detray/propagator/actors/aborters.hpp"
-#include "detray/propagator/actors/parameter_resetter.hpp"
-#include "detray/propagator/actors/parameter_transporter.hpp"
-#include "detray/propagator/navigator.hpp"
-#include "detray/propagator/propagator.hpp"
-#include "detray/propagator/rk_stepper.hpp"
+#include "detray/masks/unbounded.hpp"
 #include "detray/tracks/tracks.hpp"
-#include "tests/common/tools/inspectors.hpp"
-
-// Covfie include(s).
-#include <covfie/core/field.hpp>
-#include <covfie/core/field_view.hpp>
-
-// Vecmem include(s).
-#include <vecmem/memory/host_memory_resource.hpp>
+#include "detray/utils/axis_rotation.hpp"
+#include "tests/common/tools/intersectors/helix_plane_intersector.hpp"
 
 // google-test include(s).
 #include <gtest/gtest.h>
 
-#include <iostream>
+// System include(s)
+#include <limits>
 
 using namespace detray;
 
-namespace env {
+using matrix_operator = standard_matrix_operator<scalar>;
+using transform3 = __plugin::transform3<scalar>;
+using vector3 = typename transform3::vector3;
 
-// Type Definitions
-template <typename mask_shape_t>
-using tel_detector_t =
-    detector<detector_registry::template telescope_detector<mask_shape_t>,
-             covfie::field>;
+// Setups
+constexpr const scalar tolerance = scalar(5e-4);
+const vector3 y_axis{0.f, 1.f, 0.f};
+const vector3 z_axis{0.f, 0.f, 1.f};
+const vector3 B{0.f, 0.f, 1.f * unit<scalar>::T};
 
-constexpr scalar tol{1e-2f};
+// Algebra objects
+constexpr const cartesian2<transform3> c2;
+constexpr const helix_plane_intersector<transform3> hpi;
 
-// Surface material
-const material<scalar> mat = silicon<scalar>();
-const scalar thickness{2.f * unit<scalar>::mm};
+/// Test the path correction on a rectangular surface (cartesian coordinates)
+TEST(path_correction, cartesian2D) {
```

Comment:
So far this only reproduces the test for rectangles, right?

Reply:
Only for rectangle in this PR. (I think I will add polar and cylinder in this PR as well)  And it is more than reproducing - it was just a single surface which was not very ideal for this test